com.microsoft.mrtk.input/Experimental/SpatialMouse/Interactor/SpatialMouseInteractorCursorVisual.cs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using UnityEngine;
+using UnityEngine.Rendering;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.Input.Experimental
+{
+    /// <summary>
+    /// The cursor visual for a spatial mouse interactor. This behavior takes care of
+    /// positioning the cursor and hiding it when the mouse is not in use. 
+    /// </summary>
+    /// <remarks>
+    /// This is an experimental feature. This class is early in the cycle, it has 
+    /// been labeled as experimental to indicate that it is still evolving, and 
+    /// subject to change over time. Parts of the MRTK, such as this class, appear 
+    /// to have a lot of value even if the details haven�t fully been fleshed out. 
+    /// For these types of features, we want the community to see them and get 
+    /// value out of them early enough so to provide feedback. 
+    /// </remarks>
+    public class SpatialMouseInteractorCursorVisual : BaseReticleVisual
+    {
+        [SerializeField, Experimental]
+        [Tooltip("The ray interactor which this visual represents.")]
+        private SpatialMouseInteractor mouseInteractor;
+
+
+        [SerializeField]
+        [Tooltip("The default distance of the reticle (cursor)")]
+        private float defaultDistance = 1.0f;
+        
+        /// <summary>
+        /// See <see cref="MonoBehaviour"/>.
+        /// </summary>
+        protected void OnEnable()
+        {
+            mouseInteractor.selectEntered.AddListener(LocateTargetHitPoint);
+
+            Application.onBeforeRender += OnBeforeRenderCursor;
+        }
+
+        /// <summary>
+        /// See <see cref="MonoBehaviour"/>.
+        /// </summary>
+        protected void OnDisable()
```

Comment:
make `OnDisable()` private.

Reply:
We'd been moving to using `protected` over `private` for Unity-powered methods so they're not accidentally hidden for derived classes. I'd recommend not changing this.

https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/main/doc/UNT0021.md

Reply:
Thanks Kurtis for pointing to this information.    Given https://github.com/microsoft/Microsoft.Unity.Analyzers/blob/main/doc/UNT0021.md ,  I aggree should be made protect.

But @keveleigh ... how does this help a derived class not hide the base behaviour?  The base class would need to mark the function as `protect virtual` for the derived class to not hide the base behaviour...correct?

Reply:
You're right that it doesn't prevent the derived class from hiding the base implementation, but it at least allows the compiler to complain that you _are_ hiding the base implementation. If they're both `private`, the compiler is fine with it, but (silently) only the derived class' method gets called. You're correct on needing to mark it `virtual` if we want to allow overriding, though I don't know if we necessarily do in _all_ cases.

For example:
![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/3580640/9a40498e-9e41-4e8c-afdc-b89d3d4a9df0)

![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/3580640/837d8713-acf2-4499-b09c-ccf16b69058a)

Reply:
Got it....making this `protected` is about exposing the error/warning message.

I think I would rather do `protected virtual` to allow for extensions.  Just because it "feels" better than just making it `protected`.   Having a protected non-virtual method, in non-Unity circumstances, invites derived classes to call this function and not extended or override it.  Since an app shouldn't be calling these "Unity event functions", making it `protected virtual` at least expressions a different intent...you can extend or override the function's behavior.